packages/react-component-library/src/components/TopLevelNavigation/SheetButton.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import classNames from 'classnames'
+
+export interface SheetButtonProps extends ComponentWithClass {
+  children?: React.ReactElement
+  icon: React.ReactElement
+  onClick?: (event: React.SyntheticEvent) => void
```

Comment:
Maybe we can be more explicit about this event type?

Maybe `React.MouseEvent<HTMLButtonElement>`